### PR TITLE
TRUNK-5810 : Relationship Domain - Switching from Hibernate Mappings to Annotations

### PR DESCRIPTION
**Pull request by ManojLL**
## Description of what I changed

replace hibernate xml mappings with annotations and removed relevant xml file. 
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5810

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by ManojLL** [comment]
@dkayiwa Could you please review this PR?

File: api/src/main/java/org/openmrs/Relationship.java [hunk 1]
```diff
@@ -9,27 +9,53 @@
...
+@Table(name = "relationship")
 public class Relationship extends BaseChangeableOpenmrsData {
 	
 	public static final long serialVersionUID = 323423L;
 	
 	// Fields
-	
+	@Id
+	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "relationship_id_seq")
+	@GenericGenerator(
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Relationship.java:37]: What is the replacement for this? `unsaved-value="0"`

Comment (ManojLL) [reply api/src/main/java/org/openmrs/Relationship.java:37]: this is handled by default on hibernate. isn't it?

File: api/src/main/java/org/openmrs/Relationship.java [hunk 2]
```diff
@@ -9,27 +9,53 @@
...
+@Entity
+@Table(name = "relationship")
 public class Relationship extends BaseChangeableOpenmrsData {
 	
 	public static final long serialVersionUID = 323423L;
 	
 	// Fields
-	
+	@Id
+	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "relationship_id_seq")
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Relationship.java:36]: Could there be any reason why you did not use `GenerationType.IDENTITY`?

Comment (ManojLL) [reply api/src/main/java/org/openmrs/Relationship.java:36]: @dkayiwa  The earlier implementation utilized GenerationType.SEQUENCE, and this choice is also reflected in [this pull request](https://github.com/openmrs/openmrs-core/pull/4388/files#72), which I used as a reference. Some of the code reviews suggested a modification from GenerationType.IDENTITY to GenerationType.SEQUENCE. 

 Please let me know, if I need to change `sequence` to `Identity`.

File: api/src/main/java/org/openmrs/Relationship.java (hunk 1 above)
Comment (ManojLL) [reply api/src/main/java/org/openmrs/Relationship.java:37]: @dkayiwa  any update on this?

**Pull request merged**